### ISSUE TITLE
[RFE] IAM-linked Service Accounts

### ISSUE THREAD

**Issue by EronWright** [operator-framework/operator-lifecycle-manager#1931]
## Feature Request

**Is your feature request related to a problem? Please describe.**
In certain clouds, is possible to associate a Kubernetes Service Account (KSA) with an IAM Role/Account.  This allows the pod to assume the IAM role automatically, e.g. to interact with a cloud service such as S3 or Stackdriver.  In Google Kubernetes Engine (GKE), this is referred to as [Workload Identity](https://cloud.google.com/blog/products/containers-kubernetes/introducing-workload-identity-better-authentication-for-your-gke-applications).  In Amazon EKS, it is [IAM roles for service accounts](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html).

Both options rely on the use of an annotation on the KSA.  For example:
```yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  annotations:
    eks.amazonaws.com/role-arn: arn:aws:iam::<AWS_ACCOUNT_ID>:role/<IAM_ROLE_NAME>
```

The purpose of this issue is to add support into OLM for service accounts with IAM annotations.

**Describe the solution you'd like**
I would like a solution which allows the service account to be configurable, since the service account name typically varies across environments.  Observe the variable `AWS_ACCOUNT_ID` in the above example.   One solution might be to extend the `config` section in `Subscription` type ([ref](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/design/subscription-config.md)), to accept annotations for the service accounts that are defined in the CSV.

Note that additional configuration steps are typically needed on the Cloud IAM side, to set a policy which allows the KSA to assume the IAM role in question.  These steps generally assume that the KSA has a well-known namespace + name.

**Comment by jeloba** [comment]
I think #1881 could provide the "alternative solution."

**Comment by EronWright** [comment]
@jeloba I respectfully disagree that #1881 addresses the issue.  The alternate solution would require that OLM preserve annotations on existing service accounts, at minimum.

**Comment by jeloba** [comment]
:disappointed: You are right, didn't look into the code and expected them to improve SA handling/ownership during both creation and deletion...
